Libraries/LibGUI/GSlider.h

```diff
@@ -18,7 +18,6 @@ class GSlider : public GWidget {
     void set_max(int max) { set_range(min(), max); }
 
     int track_height() const { return 2; }
-    int knob_width() const { return 8; }
```

Comment:
I think the proportional knob width should be an optional mode for `GSlider` rather than the only behavior.
Perhaps with an API like this:

```cpp
class GSlider {
public:
    enum class KnobSizeMode { Fixed, Proportional };
    void set_knob_size_mode(KnobSizeMode mode) { m_knob_size_mode = mode; }
    KnobSizeMode knob_size_mode() const { return m_knob_size_mode; }
private:
    KnobSizeMode m_knob_size_mode { KnobSizeMode::Fixed };
};
```

If the knob size mode is `KnobSizeMode::Fixed`, it'd just do things the old way, where it's always `8px` wide.